pkg/plugin/vector/embed/embedder.go

```diff
@@ -16,9 +16,23 @@ type Embedder interface {
 	Embed(ctx context.Context, model string, text string) ([]float32, error)
 }
 
+type SecureJsonData struct {
+	BasicAuthPassword string `json:"basicAuthPassword"`
+	AuthToken         string `json:"authToken"`
+}
+
+type AuthSettings struct {
+	BasicAuthUser  string         `json:"basicAuthUser"`
+	SecureJsonData SecureJsonData `json:"secureJsonData"`
```

Comment:
Any secure fields definitely need to not be serialized as part of the `Settings` struct, otherwise they end up exposed to the frontend on /api/plugins/grafana-llm-app/settings:

```json
  "jsonData": {
    "vector": {
      "embed": {
        "type": "openai"
      },
      "enabled": true,
      "model": "text-embedding-ada-002",
      "store": {
        "authSettings": {
          "basicAuthUser": "aaa",
          "secureJsonData": {
            "basicAuthPassword": "secret"
          },
          "secureJsonFields": {}
        },
        "authType": "basic-auth",
        "type": "grafana/vectorapi"
      }
    }
  },
  "secureJsonFields": {
    "openAIKey": true
  }
```

Instead they should be stored in the separate secure JSON data object (the `secrets` field passed down to `newGrafanaVectorAPI`.